Ginger/Ginger/ExternalConfigurations/GingerAnalyticsConfigurationPage.xaml

```diff
@@ -0,0 +1,67 @@
+                        <Label x:Name="xGAAURLLabel" Content="Ginger Analytics Account URL:"   Style="{StaticResource @InputFieldLabelStyle}"  VerticalAlignment="Center" FontSize="12"/>
+                        <Label x:Name="xGAAURLLabelValidation" Content="*"   Style="{StaticResource @InputFieldLabelStyle}" VerticalAlignment="Center" Foreground="Red" FontWeight="Bold" FontSize="12"/>
+                    </StackPanel>
+                    <Activities:UCValueExpression x:Name="xGAAURLTextBox" Grid.Column="1" Grid.Row="0" HorizontalAlignment="Left" VerticalAlignment="Center" ToolTip="Ginger Analytics Account URL" Margin="10,0,0,0"  Width="400"/>
+
+                    <StackPanel Orientation="Horizontal" Grid.Column="0" Grid.Row="1">
+                        <Label x:Name="xISURLLabel" Content="Identity Service URL:" Style="{StaticResource @InputFieldLabelStyle}" VerticalAlignment="Center"  FontSize="12"/>
+                        <Label x:Name="xISURLLabelValidation" Content="*" Style="{StaticResource @InputFieldLabelStyle}" VerticalAlignment="Center" Foreground="Red" FontWeight="Bold" FontSize="12"/>
+                    </StackPanel>
+                    <Activities:UCValueExpression x:Name="xISURLTextBox" Grid.Column="1" Grid.Row="1" HorizontalAlignment="Left" VerticalAlignment="Center" ToolTip="Identiry Service URL" Margin="10,0,0,0"  Width="400"/>
+
+                    <StackPanel Orientation="Horizontal" Grid.Column="0" Grid.Row="2">
+                        <Label x:Name="xClientIdLabel" Content="Client Id:" Style="{StaticResource @InputFieldLabelStyle}" VerticalAlignment="Center"  FontSize="12"/>
+                        <Label x:Name="xClientIdLabelValidation" Content="*" Style="{StaticResource @InputFieldLabelStyle}" VerticalAlignment="Center" Foreground="Red" FontWeight="Bold" FontSize="12"/>
+                    </StackPanel>
+                    <Activities:UCValueExpression x:Name="xClientIdTextBox" Grid.Column="1" Grid.Row="2" HorizontalAlignment="Left" VerticalAlignment="Center" ToolTip="Client Id" Margin="10,0,0,0"  Width="400" LostKeyboardFocus="xClientIdTextBox_LostKeyboardFocus"/>
+
+                    <StackPanel Orientation="Horizontal" Grid.Column="0" Grid.Row="3">
+                        <Label x:Name="xClientSecretLabel" Content="Client Secret:" Style="{StaticResource @InputFieldLabelStyle}" VerticalAlignment="Center"  FontSize="12"/>
+                        <Label x:Name="xClientSecretValidation" Content="*" Style="{StaticResource @InputFieldLabelStyle}" VerticalAlignment="Center" Foreground="Red" FontWeight="Bold" FontSize="12"/>
+                    </StackPanel>
+                    <Activities:UCValueExpression x:Name="xClientSecretTextBox" Grid.Column="1" Grid.Row="3" HorizontalAlignment="Left" VerticalAlignment="Center" ToolTip="Client Secret" Margin="10,0,0,0" Width="400" LostKeyboardFocus="xClientSecretTextBox_LostKeyboardFocus"/>
```

Comment:
Try to make it generic in UCValueExpression user control itself

Reply:
Making it generic will make every field as encrypted.